### PR TITLE
autotools: fixing emitted warnings due to -no-pie detection for gcc 4.4

### PR DESCRIPTION
**Pull request by furiel**
This patchset contains the following changes:
- fixing no-pie detection for gcc 4.4. With version 4.4, gcc cannot be configured to fail on unknown cli parameters. As a result, `-no-pie gcc: error: unrecognized command line option '-no_pie'` is emitted during the build. This did not cause any problem, just an unnecessary warning is emitted.
- Adding an assert for the exciting cpu-cache optimization introduced in  9fc86dfc6de3ad66862218522b2588d4553c70de.
- using glib macro in format string to write uint64 number.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: m4/ax_compare_version.m4 [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# ===========================================================================
+#    https://www.gnu.org/software/autoconf-archive/ax_compare_version.html
+# ===========================================================================
```
Comment (MrAnno) [inline m4/ax_compare_version.m4]: Why did you copy `ax_compare_version.m4` and `ax_compiler_vendor.m4` into our repository?
`autoconf-archive` is our hard dependency. Are these macros outdated in distro's official packages?

Comment (furiel) [reply m4/ax_compare_version.m4]: I wanted to be sure these macros will be available independently from autoconf-archive version. Though most probably you are right, these are very basic macros, this has low risk. Let me check if we can compile without them.

Comment (MrAnno) [reply m4/ax_compare_version.m4]: (I've only checked with latest package in Xenial.)

**Comment by furiel** [comment]
@MrAnno I could build syslog-ng manually without hardcoded ax macros.

Though there is a little inconvenience with the release tarball generation. The process is: tarball generation executes autogen, which will steal the scripts from the host machine, and those versions of the scripts will be put in the release tarball. In the end, the content of the release tarball depends on the environment where the release tarball is created.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@furiel Yes, that is correct. Our configure script generation completely depends on the environment where the tarball is created because of `autoconf` and `autoconf-archive`.

We could get rid of at least `autoconf-archive`, but then all other macros should be copied into our repo (we also use other macros from this package).

**Pull request merged**